src/main/java/edu/cornell/kfs/pdp/service/impl/CuPaymentFileValidationServiceImpl.java

```diff
@@ -216,11 +221,34 @@ protected void checkPaymentGroupPropertyMaxLength(PaymentGroup paymentGroup, Mes
         }
     }
 
+    // This is an altered copy of checkPaymentGroupPropertyMaxLength() that operates on Payment Details instead.
+    protected void checkPaymentDetailPropertyMaxLength(PaymentDetail paymentDetail, MessageMap errorMap) {
+        for (String propertyName : paymentDetailPropertiesToCheckMaxLength) {
+            String propertyValue = (String) ObjectUtils.getPropertyValue(paymentDetail, propertyName);
+            if (StringUtils.isNotEmpty(propertyValue)) {
+                Integer maxLength = dataDictionaryService.getAttributeMaxLength(PaymentDetail.class, propertyName);
+                if ((maxLength != null) && (maxLength < propertyValue.length())) {
```

Comment:
if max length isn't set, should we log it as an error condition?

Reply:
The not-null setup here was how the related base code was set up, but I agree that some extra error handling would be good to add here if the length is null. I'll update this to throw an error if the max length is not specified, since such a situation would indicate a more significant problem with the system.

Reply:
After further discussion with Jay, I will update this to just use the Logger to log the no-max-length situation, rather than throwing an error.